presto-spark-base/src/main/java/com/facebook/presto/spark/planner/PrestoSparkRddFactory.java

```diff
@@ -232,7 +232,7 @@ private Partitioner createPartitioner(Session session, PartitioningHandle partit
         if (partitioning.equals(SINGLE_DISTRIBUTION)) {
             return new PrestoSparkPartitioner(1);
         }
-        if (partitioning.equals(FIXED_HASH_DISTRIBUTION)) {
+        if (partitioning.equals(FIXED_HASH_DISTRIBUTION) || partitioning.equals(SOURCE_DISTRIBUTION)) {
```

Comment:
Need to investigate how to appropriately set partition count for a stage that 1) scans a table. and 2) received hash partitioned results from previous stage. 
e.g. `SELECT orderkey, 1 FROM tpch.orders UNION ALL SELECT orderkey, count(*) FROM tpch.orders GROUP BY 1;`